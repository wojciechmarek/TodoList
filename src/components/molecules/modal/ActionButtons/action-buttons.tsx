import { ModalActionButtonsProps } from './action-buttons.interface';
import {
  ModalActionButtonsContainer,
  ModalCancelButton,
  ModalAcceptButton,
} from './action-buttons.styled';

export const ModalActionButtons = ({
  label,
  handleCancelButtonClick,
  handleAcceptButtonClick,
}: ModalActionButtonsProps) => {
  return (
    <ModalActionButtonsContainer>
      <ModalCancelButton onClick={handleCancelButtonClick}>
        {label.cancelButton}
      </ModalCancelButton>
      <ModalAcceptButton onClick={handleAcceptButtonClick}>
        {label.acceptButton}
      </ModalAcceptButton>
    </ModalActionButtonsContainer>
  );
};
