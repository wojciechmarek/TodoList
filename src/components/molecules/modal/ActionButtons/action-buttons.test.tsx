import { configure, render } from '@testing-library/react';
import { ModalActionButtons } from './action-buttons';
import { ModalActionButtonsProps } from './action-buttons.interface';
import '@testing-library/jest-dom';
import TestRenderer from 'react-test-renderer';

describe('ActionButtons', () => {
  let modalActionButtonsProps: ModalActionButtonsProps;

  beforeEach(() => {
    configure({ throwSuggestions: true });

    modalActionButtonsProps = {
      handleAcceptButtonClick: jest.fn(),
      handleCancelButtonClick: jest.fn(),
      label: {
        acceptButton: 'acceptButton',
        cancelButton: 'cancelButton',
      },
    };
  });

  it('should render', () => {
    const { baseElement } = render(
      <ModalActionButtons {...modalActionButtonsProps} />
    );
    expect(baseElement).toBeTruthy();
  });

  it('should match the snapshot', () => {
    const tree = TestRenderer.create(
      <ModalActionButtons {...modalActionButtonsProps} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('when the modal action buttons are rendered', () => {
    it('should render 2 buttons', () => {
      const { getAllByRole } = render(
        <ModalActionButtons {...modalActionButtonsProps} />
      );
      const buttons = getAllByRole('button');
      expect(buttons.length).toBe(2);
    });

    it('should render the accept button', () => {
      const { getByRole } = render(
        <ModalActionButtons {...modalActionButtonsProps} />
      );
      const button = getByRole('button', { name: /acceptButton/i });
      expect(button).toBeTruthy();
    });

    it('should render the cancel button', () => {
      const { getByRole } = render(
        <ModalActionButtons {...modalActionButtonsProps} />
      );
      const button = getByRole('button', { name: /cancelButton/i });
      expect(button).toBeTruthy();
    });
  });

  describe('when the button is clicked ', () => {
    it('should call the handleAcceptButtonClick function', () => {
      const { getByRole } = render(
        <ModalActionButtons {...modalActionButtonsProps} />
      );
      const button = getByRole('button', { name: /acceptButton/i });
      button.click();
      expect(
        modalActionButtonsProps.handleAcceptButtonClick
      ).toHaveBeenCalled();
    });

    it('should call the handleCancelButtonClick function', () => {
      const { getByRole } = render(
        <ModalActionButtons {...modalActionButtonsProps} />
      );
      const button = getByRole('button', { name: /cancelButton/i });
      button.click();
      expect(
        modalActionButtonsProps.handleCancelButtonClick
      ).toHaveBeenCalled();
    });
  });
});
