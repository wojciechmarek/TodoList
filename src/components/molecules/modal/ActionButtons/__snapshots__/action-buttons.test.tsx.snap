// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionButtons should match the snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-evenly;
  -ms-flex-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  justify-content: space-evenly;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 3em;
  margin: 0 2em 2em 2em;
  width: calc(100% - 4em);
}

.emotion-1 {
  cursor: pointer;
  -webkit-transition: filter 0.3s ease-in-out;
  transition: filter 0.3s ease-in-out;
  background-color: var(--add-new-task-button-bg-color);
  color: var(--button-font-color);
  border-radius: var(--standard-border-radius);
  height: 100%;
  width: calc((100% / 3) - 0.5em);
  font-weight: 900;
  font-size: 1rem;
  border: var(--standard-border);
  box-shadow: var(--button-box-shadow);
}

.emotion-1:hover {
  -webkit-filter: brightness(0.9);
  filter: brightness(0.9);
}

<div
  className="emotion-0"
>
  <button
    className="emotion-1"
    onClick={[MockFunction]}
  >
    cancelButton
  </button>
  <button
    className="emotion-1"
    onClick={[MockFunction]}
  >
    acceptButton
  </button>
</div>
`;
